smoke-tests/run_smoke_test.sh

```diff
@@ -86,19 +86,37 @@ has_intercept_id() {
 }
 
 # Puts preview url in a variable
-# Not currently used bc curl doesn't work with preview url bc we need
-# cookie system A adds (I think)
 get_preview_url() {
-    local regex="Preview URL : (https://[^ >]+)"
-    if [[ $output =~ $regex ]]; then
-        # shellcheck disable=SC2034
-        previewurl="${BASH_REMATCH[1]}"
-    else
+    preview_url=$(echo "$output" | grep -Eo 'https://[^ >]+')
+    if [[ -z $preview_url ]]; then
         echo "No Preview URL found"
         exit 1
     fi
 }
 
+# Puts workstation api key in a variable
+get_workstation_apikey() {
+    local cache_file
+    uname=$(uname)
+    if [ "$uname" == "Darwin" ]; then
+        cache_file="$HOME/Library/Caches/telepresence/apikeys.json"
+    elif [ "$uname" == "Linux" ]; then
+        if [ -n "$XDG_CONFIG_HOME" ]; then
+            cache_file="$XDG_CONFIG_HOME/telepresence/apikeys.json"
+        else
+            cache_file="$HOME/.cache/telepresence/apikeys.json"
+        fi
+    fi
+    apikey=$(grep -o -E "\"telepresence:workstation\":\"[A-Za-z0-9=]+\"" "$cache_file" | awk -F: '{print $3}')
+    if [[ -z $apikey ]]; then
+        echo "No apikey found"
+        exit 1
+    fi
+    # remove the leading and trailing "
+    apikey="${apikey%\"}"
+    apikey="${apikey#\"}"
+}
```

Comment:
I'd be inclined to just use `jq` instead of parsing JSON with awk and Bash, but I guess we don't currently have a dependency on `jq`.

I'd have no objection to adding a dependency on `jq` for this.

```sh
jq -r '.[]|.["telepresence:workstation"]|strings' <"$cache_file"
```
aught to do it

Reply:
Ah okay I also wanted to use `jq` but didn't because it wasn't already a dependency, but I think it's worth it for cleaner code that is easier to read.